db/sqlc/credential_transactions.go

```diff
@@ -98,24 +97,20 @@ func (store *SQLStore) EditCredentialTransaction(ctx context.Context, args EditC
 			return err
 		}
 
-		// Edit field records
-		for _, userFields := range args.EditFields {
-			for _, field := range userFields.Fields {
-				err = q.EditField(ctx, EditFieldParams{
-					ID:         field.ID,
-					FieldName:  field.FieldName,
-					FieldValue: field.FieldValue,
-					FieldType:  field.FieldType,
-					UpdatedBy:  uuid.NullUUID{UUID: args.EditedBy, Valid: true},
-				})
-				if err != nil {
-					return err
-				}
+		// Delete existing field records
+		for _, userFields := range args.UserFields {
+
+			err = q.DeleteFields(ctx, DeleteFieldsParams{
+				CredentialID: args.CredentialID,
+				UserID:       userFields.UserID,
+			})
+			if err != nil {
+				return err
 			}
 		}
 
 		// Create field records
-		for _, userFields := range args.AddFields {
+		for _, userFields := range args.UserFields {
```

Comment:
when you change the createdBy, wont it overwrite who created the credential

Reply:
since we are removing the old field itself, the created by field makes some sense.

The created_by and updated_by for Field table was created to accommodate in-place update. Since we decided to do a replacement instead of in-place update, those fields does not make much sense. Did not want to remove them prematurely. But we can remove if we want 

To track who updated the credential, another updated_by field is present in the credential table.